src/os/bluestore/BlueFS.cc

```diff
@@ -2192,11 +2192,11 @@ void BlueFS::compact_log_async_dump_metadata_NF(bluefs_transaction_t *t,
     std::lock_guard fl(file_ref->lock);
     if (file_ref->dirty_seq < capture_before_seq) {
       dout(20) << __func__ << " op_file_update " << file_ref->fnode << dendl;
-      t->op_file_update(file_ref->fnode);
     } else {
-      dout(20) << __func__ << " skipping just modified, dirty_seq="
+      dout(20) << __func__ << " op_file_update just modified, dirty_seq="
```

Comment:
You are correct, and I am sad.
I wanted to make async compaction as devoid of locks as possible, but it just seems to impossible to avoid one huge locking section for streaming all metadata to log.

Reply:
Can we leave it as-is for now?
I think that after we get incremental update in bluefs, less locking in streaming metadata to log might be possible.

Reply:
I have not even attempted it yet, so I do not know if I can manage it.
And I am not eager to attempt it right now.
Anyway I believe 2 things could be implemented when this is merged:
- less locking in streaming metadata
- ability to never hit "bluefs log runway exhausted"